src/dbnode/namespace/types.go

```diff
@@ -357,3 +365,21 @@ type DownsampleOptions struct {
 	// (e.g. rollup/recording rules).
 	All bool
 }
+
+// Status is the status of the namespace.
+type Status uint8
```

Comment:
nit: `NamespaceStagingStatus`

Reply:
Renamed this `StagingStatus` instead to avoid the stutter with the package name.